src/node/node_state.h

```diff
@@ -563,9 +563,18 @@ namespace ccf
         {
           LOG_DEBUG_FMT(
             "Read signature at {} for term {}", ledger_idx, last_sig->term);
-          for (auto i = term_history.size(); i <= last_sig->term; ++i)
+          // Initial transactions, before the first signature, must have
+          // happened in the first signature's term (eg - if the first
+          // signature is at idx 20 in term 4, then transactions 1->19 must
+          // also have been in term 4). The brief justification is that while
+          // the first node may start in an arbitrarily high term (it does not
+          // necessarily start in term 1), it cannot _change_ term before a
+          // valid signature.
```

Comment:
Another way to look at this is that the first term change can only happen in a network of more than one node. For the network to have more than one node, there must have been at least one successful config change, which means at least one commit, which means at least one signature.